source/plugins/go/src/telemetry.go

```diff
@@ -556,17 +556,27 @@ func PushToAppInsightsTraces(records []map[interface{}]interface{}, severityLeve
 		} else if strings.Contains(logEntry, "E! [inputs.prometheus]") {
 			populateKubeMonAgentEventHash(record, PromScrapingError)
 		} else if strings.Contains(logEntry, "Lifetime validation failed. The token is expired.") {
-			UpdateMdsdErrorMetrics("MdsdTokenExpired")
+			UpdateTracesErrorMetrics("MdsdTokenExpired")
 		} else if strings.Contains(logEntry, "Failed to upload to ODS: Error resolving address") {
-			UpdateMdsdErrorMetrics("MdsdODSUploadErrorResolvingAddress")
+			UpdateTracesErrorMetrics("MdsdODSUploadErrorResolvingAddress")
 		} else if strings.Contains(logEntry, "Data collection endpoint must be used to access configuration over private link") {
-			UpdateMdsdErrorMetrics("MdsdPrivateLinkNoDCE")
+			UpdateTracesErrorMetrics("MdsdPrivateLinkNoDCE")
 		} else if strings.Contains(logEntry, "Failed to register certificate with OMS Homing Service:Error resolving address") {
-			UpdateMdsdErrorMetrics("MdsdOMSHomingServiceError")
+			UpdateTracesErrorMetrics("MdsdOMSHomingServiceError")
 		} else if strings.Contains(logEntry, "Could not obtain configuration from") {
-			UpdateMdsdErrorMetrics("MdsdGetConfigError")
+			UpdateTracesErrorMetrics("MdsdGetConfigError")
 		} else if strings.Contains(logEntry, " Failed to upload to ODS: 403") {
-			UpdateMdsdErrorMetrics("MdsdODSUploadError403")
+			UpdateTracesErrorMetrics("MdsdODSUploadError403")
+		} else if strings.Contains(logEntry, "failed getting access token") {
+			UpdateTracesErrorMetrics("AddonTokenAdapterFailedGettingAccessToken")
+		} else if strings.Contains(logEntry, "failed to watch token secret") {
+			UpdateTracesErrorMetrics("AddonTokenAdapterFailedToWatchTokenSecret")
+		} else if strings.Contains(logEntry, "http: Server closed") {
+			UpdateTracesErrorMetrics("AddonTokenAdapterServerClosed")
+		} else if strings.Contains(logEntry, "forwarding the token request to IMDS...") {
+			UpdateTracesErrorMetrics("AddonTokenAdapterForwardingTokenRequestToIMDS")
+		} else if strings.Contains(logEntry, "watch channel is closed, retrying..") {
+			UpdateTracesErrorMetrics("AddonTokenAdapterWatchChannelClosed")
```

Comment:
will we drop the container logs from addon token adaptor pod now that we converting useful logs to metrics?

Reply:
yes they wont be showing up in traces